Update documentation to add pixi.toml and to reflect move away from robostack-staging channel (#73)
RoboStack/robostack.github.io: docs/GettingStarted.md

# Getting Started

> RoboStack is a bundling of ROS for Linux, Mac and Windows using the [Conda package manager](https://docs.conda.io/en/latest/), based on top of [conda-forge](https://conda-forge.org/).

Reviewer: ```suggestion
RoboStack is a bundling of ROS for Linux, macOS and Windows using the [Conda package manager](https://docs.conda.io/en/latest/), based on top of [conda-forge](https://conda-forge.org/).
```

nitpick: the operating system name is `macOS`, not `Mac`. However, this is inconsistent with the rest of operating system names (`Windows` and `Linux`) so it is matter of taste, so feel free to ignore this comment.

Author: This was the existing language in the docs, but I agree that it's better to be explicit. Do you think it's worth mentioning in that same heading block that both intel and arm-based Macs are supported?

        
    !!! warning "Do not source the system ROS environment"
        When there is an installation available of ros on the system, in non-conda environments, there will be interference with the environments. As the `PYTHONPATH` set in the setup script conflicts with the conda environment.

    !!! warning "PowerShell is not supported"

        On Windows, Powershell is not supported, only the Command Prompt terminal is supported.
    
    ## Installing ros
    === "Mamba"

        ```bash title="Prepare an environment to use the correct channels"
        mamba create -n ros_env python=3.11
        mamba activate ros_env

        # this adds the conda-forge channel to the new created environment configuration 
        conda config --env --add channels conda-forge
        # remove the defaults channel just in case, this might return an error if it is not in the list which is ok
        conda config --env --remove channels defaults
        ```
        !!! note

>             There are different channels depending on the version of ROS that you want to install, you must only install ONE version of ROS per environment:

Reviewer: Note that I am not sure if this is entirely true, as you may want to install a distro of ROS 1 and ROS 2 in the same environment to use the ROS 1 <--> ROS 2 bridges, but for a getting started guide probably this message is sufficient.

Author: ```suggestion
            There are different channels depending on the version of ROS that you wish to install, to add these channels and install your desired version, you can run the following:
```

Reviewer: Great!

        === "ROS1 Noetic"
            ```
            conda config --env --add channels robostack-noetic
            mamba install ros-noetic-desktop
            ```
        === "ROS2 Humble"
            ```
            conda config --env --add channels robostack-humble
            mamba install ros-humble-desktop
            ```
        === "ROS2 Jazzy"
            ```
            conda config --env --add channels robostack-jazzy
            mamba install ros-jazzy-desktop
            ```

    
        ```bash title="Deactivate and reactivate the environment to initialize the configured ROS environment"
        mamba deactivate
        mamba activate ros_env
        ```

    === "Micromamba"

        When you only have micromamba available use the following commands: 

        === "ROS1 Noetic"
            ```
            # Create a ros-noetic desktop environment

>             micromamba create -n ros_env -c conda-forge -c robostack-noetic ros-noetic-desktop python=3.11

Reviewer: ```suggestion
            micromamba create -n ros_env -c conda-forge -c robostack-noetic ros-noetic-desktop
```

Perhaps we can remove this for consistency with humble and jazzy? As the env in this case is created in one shot (differently from the mamba instructions), there is no reason to specify before hand the python version, and specifying it explicitly here will create us problem if we ever want to update the Python version.

    
    !!! warning "Do not source the system ROS environment"
        When there is an installation available of ROS on the system, in non-conda environments, there will be interference with the environments. As the `PYTHONPATH` set in the setup script conflicts with the conda environment.

    ## Install RoboStack using Pixi

    Initialize a new project and navigate to the project directory.
    ```shell
    pixi init robostack
    cd robostack
    ```

    Open the newly created pixi.toml in your favourite text editor and paste the below configuration into the file (overwriting the configuration created by `pixi init`):
    ``` bash title="pixi.toml"
    [project]
    name = "robostack"
    version = "0.1.0"
    description = "Development environment for RoboStack ROS packages"
    authors = ["Your Name <your.email@example.com>"]
    channels = ["https://fast.prefix.dev/conda-forge"]
    platforms = ["linux-64", "win-64", "osx-64", "osx-arm64", "linux-aarch64"]

    [target.win-64.dependencies]
    #vs2022_win-64 = "*" #Uncomment if using Visual Studio 2022

    [dependencies]

>     python = "==3.11"

Reviewer: ```suggestion
    python = "*"
```

I would avoid to hardcode the Python version here. This will be annoying to update when we will update the Python version in a distro and not another. I know we are hardcoding python in the mamba instructions, but there the problem is a bit different as the environment is created with different commands, while here in pixi is always created in one shot based on the `pixi.toml`.

    compilers = "*"
    cmake = "*"
    pkg-config = "*"
    make = "*"
    ninja = "*"

    [environments]
    noetic = { features = ["noetic"] }
    humble = { features = ["humble"] }
    jazzy = { features = ["jazzy"] }

    #noetic
    [feature.noetic]
    channels = ["https://prefix.dev/robostack-noetic"]

    [feature.noetic.dependencies]
    ros-noetic-desktop = "*"
    catkin_tools = "*"
    rosdep = "*"

    #humble
    [feature.humble]
    channels = ["https://prefix.dev/robostack-humble"]

    [feature.humble.dependencies]
    ros-humble-desktop = "*"
    colcon-common-extensions = "*"
    rosdep = "*"

    #jazzy
    [feature.jazzy]
    channels = ["https://prefix.dev/robostack-jazzy"]

    [feature.jazzy.dependencies]
    ros-jazzy-desktop = "*"
    colcon-common-extensions = "*"
    rosdep = "*"
    ```
    ```bash
    #Save and exit pixi.toml
    pixi install

>     #You can now start an environment with your desired robostack distribution using one of the below commands (must be in the project directory):

Reviewer: Just FYI, you can avoid to be in the project directory if you pass the `--manifest-path` option, but I understand that explaining this may be an overkill here. Perhaps @ruben-arts has some idea of a good wording of this.

Author: ```suggestion
    #You can now start an environment with your desired robostack distribution using one of the below commands (either executed from within the project directory or my appending `--manifest-path` and pointing to your project directory):
```
TIL about `--manifest-path` ...

Reviewer: Perfect, thanks!
